tests/TestCase.php

```diff
@@ -73,7 +73,8 @@ public function mockComponent($config = []) {
      */
     protected function expectRedirect($url)
     {
-        $this->setExpectedException('\yii\base\Exception', $this->prepareUrl($url));
+        $url = $this->prepareUrl($url) ? : '/';
+        $this->setExpectedExceptionRegExp('\yii\base\Exception', '#^' . $url . '$#');
```

Comment:
Wow, that was a big surprise to me. Thanks a lot for finding this. I felt so safe using this method and did never think about that this assertion could not work as expected.